src/Mobs/Wolf.cpp

```diff
@@ -29,15 +30,37 @@ cWolf::cWolf(void) :
 
 bool cWolf::DoTakeDamage(TakeDamageInfo & a_TDI)
 {
+	cEntity * PreviousTarget = m_Target;
 	if (!super::DoTakeDamage(a_TDI))
 	{
 		return false;
 	}
 
-	if (!m_IsTame)
+
+	cPawn * Pawn = static_cast<cPawn*>(m_Target);
```

Comment:
I tested your code. The server crashed here and m_Target was nullptr. The wolf wasn't tamed,